docs/semgrep-supply-chain/upgrade-guidance.md

```diff
@@ -0,0 +1,122 @@
+- Your **private** GitHub Semgrep app must have [**Read and write** access on the **Contents** permission](#grant-read-and-write-access-to-a-private-github-semgrep-ap#grant-read-and-write-access-to-a-private-github-semgrep-appp).
+
+### Grant **Read and write** access to a private GitHub Semgrep app
+
+To change your Semgrep app's permission:
+
+1. Navigate to the settings page of your private Semgrep GitHub app; refer to [<i class="fas fa-external-link fa-xs"></i> Changing the permissions of a GitHub app](https://docs.github.com/en/apps/maintaining-github-apps/modifying-a-github-app-registration#changing-the-permissions-of-a-github-app) for instructions.
+1. In the **Repository permissions** section, search for `Contents`.
+1. Click the drop-down menu and select **Read and write**.
+
+## How it works
+
+After enabling dependency guidance, Semgrep performs post-scan analysis and marks applicable findings as **Safe to upgrade** or with **Breaking changes**.
+
+- This analysis is performed every **two hours** on the latest **full scan**.
+- Only findings whose dependencies have **fixed versions** that resolve the vulnerability are marked by Semgrep as **Safe to upgrade** or with **Breaking changes**.
+- Findings without any fixed versions have no badge; instead they say **no patch available**.
+  ![Finding with no fixed version available](/img/no-patch-available.png#md-width)
+  _**Figure**. **Details** page for a finding that has no available fix._
+
+The following chart shows the steps Semgrep performs from scanning to analysis and the actions you can take based on the advice it shows.
+
+![Flowchart explaining how Semgrep provides upgrade guidance and possible actions to take based on its advice.](/img/upgrade-guidance-flowchart.png)
+
+## Review a finding's upgrade guidance 
+
+<ViewDetailsSsc />
+
+![SSC details that provide upgrade guidance.](/img/vuln-panels-ssc.png)
+_**Figure**. Useful details that provide upgrade guidance._
+<dl>
+<dt>A - Upgrade badge</dt>
+<dd>Indicates if an upgrade is safe or may break your codebase.</dd>
+<dt>B - The line of code (LOC) of the finding</dt>
+<dd>This shows the LOC that caused the finding; this does <strong>not</strong> show the LOC where the breaking changes occur.</dd>
+<dt>C - Link to change list drawer</dt>
+<dd>Click this link to display the LOC where a breaking change may occur.</dd>
+<dt>D - Open fix PR button</dt>
+<dd>Click this button to open a PR with the code to upgrade the dependency to a safe version, if any.</dd>
+</dl>
+
+![Drawer showing all lines of code that must be changed](/img/upgrade-guidance-drawer.png#md-width)
+_**Figure**. Drawer showing all the lines of code that must be changed or are safe._
+
+## Create a pull request with fixes
+
+1. Navigate to the **Details** page of the finding for which you want to make a pull request.
+1. Click **Fix** > **Open fix PR**.
+
+A pull request includes:
+
+- The code change to upgrade the dependency
```

Comment:
I might phrase this as something more like "the manifest/lockfile change to upgrade the dependency", since we aren't making actual _code_ changes